recorder.go

```diff
@@ -104,3 +105,72 @@ func (p *Printer) Record(loc *Location, op Operation, stats *Stats) error {
 	p.lastValue = time.Now()
 	return nil
 }
+
+// Full Recorder records all of the evaluations that occur during an optimization
```

Comment:
It seems specific? In what way? It seems like a natural counterpart to Printer. Printer prints to a writer to track optimization, this saves the optimization run to `json` for retrospection